src/java.base/share/classes/java/io/ObjectInputStream.java

```diff
@@ -3670,6 +3674,23 @@ String readLongUTF() throws IOException {
          * utflen bytes.
          */
         private String readUTFBody(long utflen) throws IOException {
+            int ascii = 0;
+            if (!blkmode) {
+                end = pos = 0;
+            } else {
+                int avail = end - pos;
+                if (avail >= utflen) {
+                    ascii = JLA.countPositives(buf, pos, (int)utflen);
+                    if (ascii == utflen) {
+                        // For ASCII-only bytes we can treat the bytes as ISO-8859-1 and
+                        // avoid a redundant scan
+                        String utf = new String(buf, pos, (int)utflen, StandardCharsets.ISO_8859_1);
+                        pos += (int)utflen;
```

Comment:
```suggestion
                        String utf = new String(buf, pos, ascii, StandardCharsets.ISO_8859_1);
                        pos += ascii;
```
Redundant casts

Reply:
I get warnings and build failures if I leave out those casts (`utflen` is a `long`)

Reply:
Missed this comment, but ended up doing like you suggested.